test_agent/rust/src/main.rs

```diff
@@ -65,8 +92,20 @@ async fn connect_and_receive() -> Result<(), Box<dyn std::error::Error>> {
 #[tokio::main]
 async fn main() -> Result<(), Box<dyn std::error::Error>> {
     let handle = thread::spawn(|| {
-        let rt = Runtime::new().expect("Error creating runtime");
-        match rt.block_on(connect_and_receive()) {
+        let args: Vec<String> = env::args().collect();
```

Comment:
The `clap` crate is what I've used in the past for parsing command line arguments. It's a bit more "batteries included" and nicer to work with.

IMHO you should consider using it.

Reply:
yep, it's pretty nifty, I integrated it